### ISSUE TITLE
Clarify battery type of vehicle and storage batteries

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#42902]
# Is your feature request related to a problem? Please describe.
In order to continue expanding chemistry options and the chemistry skill, certain things that are currently abstract need to be less abstract so that we can do SCIENCE to them.

# Describe the solution you'd like
Car/truck batteries and vehicle storage batteries need to be clear on if they are lead-acid, AGM, EFB, lithium ion, or something else.

This means creating new item IDs and possibly different stats for the types of battery.

At the moment I know relatively little about the various types of car battery, so I would say that we can roughly break it into:
- car batteries that a survivor might be able to carefully dismantle, repair, or even build from scratch (archetypally, the lead-acid cell, but there may be more). Assuming there are different things a survivor could do with/to these batteries, they should be modeled separately so that we can add those actions.
- car batteries a survivor can't do much with, no matter their tools or knowledge (lithium ion is the classic example).

For the latter, if there are several types of battery cell that will fall into this category, it would perhaps be reasonable to make a lump-them-together item rather than distinguishing them. "Advanced car battery" for example. There's no point in representing a bunch of batteries that are all going to have the same stats and purpose in game.

# Additional context
The real trick here will be not expanding the number of cars in game to make multiple versions that only differ by battery. For the moment I'd suggest just giving each vehicle one specific battery subtype.

Later on we should get some code for allowing randomization of car parts, but that's a different story.

# Going forward
It's not part of this Issue but once we have modeled this we can get into the Fun Stuff you can do with lead acid batteries, giving us more good times with the chemistry skill.

**Comment by Zireael07** [comment]
From what I can tell (some net research plus my dad is a car nut, used to run a local repair shop before repairing cars became mostly a computer thing (programming the central computer, ECU, sat-nav and the like)) .... most modern (gasoline) cars and trucks use lead-acid or AGM. Motorbikes seem to use AGM because it can't spill the electrolyte.
Cata is now set in modern times, not the future, so except electric vehicles that run on some special batteries I can't figure out online, we can probably continue with the assumption that car batteries are lead-acid unless specifically called out.
The only example of li-ion car batteries I could find is for high-end sports cars, so the electric/solar sports car, if it's still in the game, could be the one exception.

**Comment by Fosheze** [comment]
This is something I have a bit of knowledge about. I currently work with portable inverter setups, have worked with electronics, and I'm kind of a car guy IRL so I do quite a bit of research on battery types. As @Zireael07 said, almost all vehicle batteries are going to be AGM, Flooded(lead acid), EFB, or Gel. The only one of those which is realistically construct-able by a survivor is the basic flooded. While I do personally like the idea if differentiating them I'm not sure if there is a real game play reason in doing so. The main differences between them (other than capacity of course) are the maximum safe amp draw, ease of recharging, longevity, and durability all of which aren't currently modeled in CDDA. They're also all "Lead-Acid" at their core. They're just manufactured in different ways for different applications so they would break down into roughly the same components. Chemically speaking, you can get the same stuff from all of them.

As far as rechargeable storage batteries go the only ones that I know of being used en mass are deep-cycle lead acid (same as the automotive batteries but with thicker plates) and the various lithium chemistries. Nickel Cadmium to my knowledge has been entirely replaced by lithium now. Unless I'm forgetting something, which is always possible, other battery types would be very niche. Unfortunately recent consumer battery tech has basically just been about different ways to use lithium. The current storage battery system also already does a good job of just having large storage batteries made up of many smaller storage batteries like is done IRL. Someone correct me if I'm wrong because I'm not a chemist but, I don't think there's much survivor usable chemical difference between the lithium chemistries.  Chemically speaking, I'm not sure there's much point in differentiating between anything other than vehicle(lead acid) and storage(lithium) like is basically already done. 

I fount a pretty decent [Wikipedia list](https://en.wikipedia.org/wiki/List_of_battery_types) but like I said most chemistries other than the lead acid and lithium are incredibly niche to the point where the average survivor probably wouldn't really find them anywhere.

**Comment by I-am-Erk** [comment]
@Fosheze the main questions I'd have for you then, if you know the answers:
- in terms of what a survivor  can do, is there much practical difference between AGM and traditional lead-acid batteries? Could a knowledgable survivor still take apart an AGM and do the same stuff with the innards?
- How common do you think lead-acid storage batteries are, should we model both types or just make them all LI?

There's definitely nothing survivor-usable inside an LI battery presently. They have a bad habit of exploding when you try to open them up to find out.

**Comment by Fosheze** [comment]
@I-am-Erk AGM stands for absorbed glass mat. The only difference between an AGM battery and a standard battery is that rather than the acid just flowing freely around the plates as a liquid, it is absorbed into a fiberglass mat pressed between the plates. This makes everything more rigid by taking up all the spare space. The plates are the same and the acid could be recovered by just wringing out the fiberglass mats. Gel batteries are similar except rather than a mat the acid is gelled, which could probably be undone but once again, I'm not a chemist.

As far as how common they are I couldn't say for sure because I haven't done any market research. I currently use deep cycle lead acid in everything I do because I ship these things and those have fewer shipping restrictions. Even then I'm strongly considering switching to lithium. Older in place inverter systems will almost always be using lead acid because that's all that was practical at the time. But now Lithium is almost always the way to go. If I was to give a completely baseless guess I would say that it's probably currently 50/50 in inverter applications with Lithium being the most common in new systems. Considering the Cataclysm is present day but more advanced in many ways it would probably be safe to just make them all lithium.

As far as your safety comment goes though, that's only really true for extremely high density lithium polymer batteries like cellphone batteries. Larger batteries used for bulk storage applications normally use much safer chemistries like Lithium Iron Phosphate so at worst you will get a fizzle and some really nasty fumes. I have cracked open old lithium cells myself for the fun of it before without issue. A proper chemist could probably make some fun stuff from the hydrogen flouride in the electrolyte or the lithium in the plates. The only real danger in most of the larger batteries is that if you do somehow manage to light them on fire you aren't going to be putting it out easily. My coworker who is in charge of our data center back up power has frequently told me that if it goes up in flames it will just need to burn it's self out because the halon can't put that out.

**Comment by eilaattwood** [comment]
> most modern (gasoline) cars and trucks use lead-acid or AGM. Motorbikes seem to use AGM because it can't spill the electrolyte.

I've disassembled AGM battery few times. Most of them are just good old lead-acid batteries, main difference from classic ones is that they have some spongy material between electrodes that is soaked with acidic electrolyte, when classic ones are just filled with liquid electrolyte. All other is the same, including shape and material of the electrodes.

**Comment by eilaattwood** [comment]
> I have cracked open old lithium cells myself for the fun of it before without issue.

That's true... unless they are charged. All stored enegry must go somewhere when the battery is damaged, and for charged bulk storage batteriy thats a LOT of energy. Can do a nice flame or even bang. That you cannot put out before all stored energy is dissipated.

**Comment by eilaattwood** [comment]
>    How common do you think lead-acid storage batteries are, should we model both types or just make them all LI?
Almost all non-electryc car batteries are lead, almost all electric car batteries are lithium, notebook batteries are lithium, cellphone and smartphone batteries are lithium, batteries from UPS and backup systems are usually lead, but also can be lithium. Disposable batteries are mixed: lithium, alkaline, or salt.

**Comment by eilaattwood** [comment]
Also, if we decide to make stock batteries non-repairable and not craftable, we should definitely add some makeshift variant for use in vehicles. With less capacity, more volume and mass, or both, but possible to craft from scratch. Because that's definitely possible.

**Comment by Zireael07** [comment]
> Because that's definitely possible.

Source, please? (I know Kevin will surely ask for one when he pops back up ;) )

**Comment by eilaattwood** [comment]
> 
> 
> > Because that's definitely possible.
> 
> Source, please? (I know Kevin will surely ask for one when he pops back up ;) )

Lead-acid battery was invented in 19 century by a single inventor. Less efficient, than modern ones, but it was usable. Silver-oxide are also not that complicated. No need of nanotechnology to make usable prototype.

**Comment by eilaattwood** [comment]
Speaking of lead acid batteries and their repairing. This is what can happen:

* Damage of the case followed by the electrolyte leakage. Repair is quite straightforward: patch up leakage, replace or refill electrolyte.
* Wrong acid to water ratio. Can often happen to open maintainable ones or AGM with valves, as water can evaporate during routine cycle or overcharge. Can be repaired by adding water or acid, tool for electrolyte density measurement is required.
* Sulfatation. Greatly decreases capacity. Result of storing the battery deeply discharged. Can possibly be cured by controled pulse charger.
* Plates shortage due to mechanical damage of plates. Very difficult to repair. In best case can be repaired by cleaning the bottom of the case and plates of lead flakes that fell off. Requires full disassembling of the battery. In worst case unrepairable, are will cost in permanent great loss of capacity to almost unusable state.

**Comment by Alex-Folts** [comment]
> * How common do you think **lead-acid storage batteries** are, should we model both types or just make them all LI?

You most likely find them in stationary energy storages, in regular house with solar panels, in remote off grid solar/wind powered house, in some buildings with backup power storage, in boat/yacht, maybe in train, i think some electric forklift use them too, everywhere where extra weight is not so important. They are cheaper than lithium, but heavier per joule of stored energy, you wont find them in brand EV/hybrids, but in some DIY converted EV can be occasionally found.

They usually be like [this](https://www.alibaba.com/product-detail/Factory-Prices-Top-Grade-Gel-Agm_60787962608.html?spm=a2700.7724857.normalList.40.121e106cQyAejF), so for ingame use we could assume that we have battery bank which consist of 4 of them(~220kg total mass, ~10-12kW*h).

**Comment by Fosheze** [comment]
> > I have cracked open old lithium cells myself for the fun of it before without issue.
> 
> That's true... unless they are charged. All stored enegry must go somewhere when the battery is damaged, and for charged bulk storage batteriy thats a LOT of energy. Can do a nice flame or even bang. That you cannot put out before all stored energy is dissipated.

That's true but it also isn't hard to drain a small lithium cell completely. If you really want to redneck it you could just use a length of fairly small wire to act as a a resistive heating element. I'm not saying that's smart but there are a lot of ways to kill a battery. In game I believe there is already a way to only allow them to be disassembled when they're empty if we wanted to actively enforce that; I seem to remember encountering that as a crafting issue with other items. As far as the larger storage batteries go though you would typically break them down into their component cells first anyways rather than trying to drain the whole thing at once. This could be modeled by simply keeping the current storage battery disassembly system where they break down into multiple smaller storage batteries.

**Comment by anothersimulacrum** [comment]
@eilaattwood I don't think anyone's opposed to lead/acid batteries being repairable and makeable to an extent.

The same cannot be said for lithium derived ones.

**Comment by Zireael07** [comment]
> The same cannot be said for lithium derived ones.

I don't think anyone was talking about DIY lithium batteries.

Also @Alex-Folts good points - my parents' yacht does have lead-acid storage batteries (and solar panels)

**Comment by actual-nh** [comment]
AGM and gel are both using glass/silica, specifically because it doesn't interact with the other components, so that's a filtration question as opposed to a chemical reaction question (unless the chemical reaction was with the other components, of course).

In terms of fires, from the research I'm finding (e.g., <https://www.nature.com/articles/s41598-017-09784-z>) putting it out with water is most decidedly going to make things worse - that HF that @Fozheze mentioned seems to actually be produced mainly from water interacting with an anhydrous electrolyte, although it's possible that this varies by type of battery. Fires from the usual Li-ion battery usually happen when something like a short circuit causes an initial temperature rise - thus rather more likely if a battery is charged - which then triggers highly exothermic reactions that only happen at higher temperatures. Most of these reactions don't require oxygen, although some of the gasses released (most notably hydrogen) can certainly cause a flame with oxygen. Most extinguisher systems, if they don't disperse the materials or chill down the temperature, may suppress any resulting oxygen flames, but won't do much if anything about the high temperatures causing a chain reaction in other battery cells (see <https://www.osti.gov/servlets/purl/1249044> plus <https://www.consumerreports.org/electronics/problem-with-stowing-lithium-ion-batteries-on-planes/> and some links from there). Moreover, halons themselves can be decomposed at the temperatures we're talking about, releasing further toxic gasses (including vaporized HCl; older halons, with fluorine, could also give you HF...). The best ones at this are class D (dry powder), although others with nonreactive (including without water!) chemicals can help.

**Comment by GuardianDll** [comment]
Technically solved by #75361, even if it is only for test purposes
